### ISSUE TITLE
New Rule: Report on and remove trailing slashes from links

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/markdown#448]
### Rule details

If a URL has a trailing link, e.g. `www.google.com/`, that trailing `/` is unnecessary and should be removed.

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
https://www.google.com/
www.google.com/
google.com/
```

### Prior Art

```markdown
* https://github.com/vhf/remark-lint-no-url-trailing-slash
* https://github.com/xiaogaozi/markdownlint-rule-no-trailing-slash-in-links
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### Additional comments

This was originally requested in #228 by @DamianGlowala. Back then, this project was just a processor and did not have rules. It does now.

**Comment by mdjermanovic** [comment]
This would apply only to trailing slashes after the domain name? Not, for example, `https://www.google.com/foo/`.

**Comment by nzakas** [comment]
:-1: I don't think this is specific enough to Markdown to be included (the same logic could be applied to links in any language). I also feel like the unpredictability of the behavior of removing slashes and the desire or not to have them create enough variables that this wouldn't be generally applicable.

As always, you can create a custom rule for this purpose if you so desire.

**Comment by lumirlumir** [comment]
@JoshuaKGoldberg Thanks for suggesting this rule :)

However, I believe there was a consensus among the team members, so I’m closing this as not planned.

---

FYI, it may take some time, but I'm planning to create a space for Markdown rules that aren't accepted into the core of the Markdown plugin but could still be useful.

If you're interested, feel free to open an issue at https://github.com/lumirlumir/npm-eslint-plugin-mark — I'd be happy to take a look!
